third_party/move/move-compiler-v2/tests/eager-pushes/eager_load_03.exp

```diff
@@ -0,0 +1,112 @@
+============ initial bytecode ================
+
+[variant baseline]
+fun m::bar($t0: &mut u64) {
+  0: return ()
+}
+
+
+[variant baseline]
+fun m::baz($t0: u64, $t1: u64) {
+  0: return ()
+}
+
+
+[variant baseline]
+public fun m::foo($t0: u64) {
+     var $t1: u64
+     var $t2: &mut u64
+  0: $t1 := m::one()
+  1: $t2 := borrow_local($t0)
+  2: m::bar($t2)
+  3: m::baz($t0, $t1)
+  4: return ()
+}
+
+
+[variant baseline]
+fun m::one(): u64 {
+     var $t0: u64
+  0: $t0 := 1
+  1: return $t0
+}
+
+============ after LiveVarAnalysisProcessor: ================
+
+[variant baseline]
+fun m::bar($t0: &mut u64) {
+     # live vars: $t0
+  0: drop($t0)
+     # live vars:
+  1: return ()
+}
+
+
+[variant baseline]
+fun m::baz($t0: u64, $t1: u64) {
+     # live vars: $t0, $t1
+  0: return ()
+}
+
+
+[variant baseline]
+public fun m::foo($t0: u64) {
+     var $t1: u64
+     var $t2: &mut u64
+     # live vars: $t0
+  0: $t1 := m::one()
+     # live vars: $t0, $t1
+  1: $t2 := borrow_local($t0)
+     # live vars: $t0, $t1, $t2
+  2: m::bar($t2)
+     # live vars: $t0, $t1
+  3: m::baz($t0, $t1)
+     # live vars:
+  4: return ()
+}
+
+
+[variant baseline]
+fun m::one(): u64 {
+     var $t0: u64
+     # live vars:
+  0: $t0 := 1
+     # live vars: $t0
+  1: return $t0
+}
+
+
+============ disassembled file-format ==================
+// Move bytecode v7
+module c0ffee.m {
+
+
+bar(Arg0: &mut u64) /* def_idx: 0 */ {
+B0:
+	0: MoveLoc[0](Arg0: &mut u64)
+	1: Pop
+	2: Ret
+}
+baz(Arg0: u64, Arg1: u64) /* def_idx: 1 */ {
+B0:
+	0: Ret
+}
+public foo(Arg0: u64) /* def_idx: 2 */ {
+L1:	loc0: u64
+B0:
+	0: Call one(): u64
+	1: MutBorrowLoc[0](Arg0: u64)
+	2: Call bar(&mut u64)
+	3: StLoc[1](loc0: u64)
+	4: MoveLoc[0](Arg0: u64)
```

Comment:
Yes, this was a test case to show that you should not pre-load `Arg0` because of the `&mut` to it, like you point out.

I should have mentioned in the PR description that there are some negative test cases as well - this one is actually handcrafted and not reduced from framework code.